internal/storage/postgres/tree/create.go

```diff
@@ -114,3 +131,25 @@ func (r *TreeRepository) linkImages(ctx context.Context, treeID, imgID int32) er
 
 	return r.store.LinkTreeImage(ctx, &params)
 }
+
+func (r *TreeRepository) validateTreeEntity(ctx context.Context, tree *entities.Tree) error {
+	if tree == nil {
+		return errors.New("tree is nil")
+	}
+	if tree.Latitude < -90 || tree.Latitude > 180 {
+		return storage.ErrInvalidLatitude
+	}
+	if tree.Longitude < -180 || tree.Longitude > 180 {
+		return storage.ErrInvalidLongitude
+	}
+	params := sqlc.GetTreeByCoordinatesParams{
+		Latitude:  tree.Latitude,
+		Longitude: tree.Longitude,
+	}
+	treeByCoordinates, err := r.store.GetTreeByCoordinates(ctx, &params)
+
+	if err == nil && treeByCoordinates != nil {
+		return storage.ErrTreeWithSameCoordinates
+	}
```

Comment:
What about the import logic we built in release one? We need the function `GetTreeByCoordinates` for the import functionality to check if a tree with the same coordinates exists. Please have a look at `\internal\service\domain\tree\import_tree.go`.

We assumed that the coordinates can be mapped to trees.

Reply:
OK, that's a point, but it's being moved to the plugin system so the plugin will have to deal with it. However, my point is still that there can be trees or sensors with the same coordinates. Just because the TBZ can't give us an ID for the trees, we shouldn't rely on it being unique. Coordinates are always more accurate depending on how many decimals you use. Another argument is, what if a tree, sensor or tree cluster is archived? Then a new object can be created with the same coordinates.

Reply:
Okay, we have to discuss these points in our weekly meeting. For now, I cannot remove the function because I need to change other points that are unrelated to the purpose of this MR (testing the tree repository). We can create another issue for it.